tonic/tonic.go

```diff
@@ -254,27 +255,35 @@ func extractQuery(c *gin.Context, tag string) (string, []string, error) {
 	if err != nil {
 		return "", nil, err
 	}
-
-	rawQ := c.Request.URL.Query()[name]
-
-	// delete empty elements so default+required will play nice together
-	// append to a new collection to preserve order without too much copying
-	q := make([]string, 0, len(rawQ))
-	for i := range rawQ {
-		if rawQ[i] != "" {
-			q = append(q, rawQ[i])
+	var params []string
+	query := c.Request.URL.Query()[name]
+
+	if c.GetBool("explode") {
+		// Delete empty elements so default and required arguments
+		// will play nice together. Append to a new collection to
+		// preserve order without too much copying.
+		params = make([]string, 0, len(query))
+		for i := range query {
+			params = append(params, query[i])
+		}
+	} else {
+		splitFn := func(c rune) bool {
+			return c == ','
+		}
+		if len(query) > 0 {
+			params = strings.FieldsFunc(query[0], splitFn)
```

Comment:
why not strings.Split() ?

Reply:
What about query[1], potentially ?

Reply:
I used `strings.FieldsFunc` with a custom function to split the input instead of `strings.Split` because the later does not ignore the empty values.

If `explode` is `false`, what do you think we should do if there is more that one occurrence of the variable in the querystring ? I used `query[0]` but we could use the last one instead.

Reply:
re: FieldsFunc, ok, good point

For explode false + multi values... is that an error case, maybe ?

Reply:
As agreed, an error is returned if a repeated variable is found while `explode` is disabled.